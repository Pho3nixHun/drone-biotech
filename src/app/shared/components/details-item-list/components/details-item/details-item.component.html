<div class="flex flex-row space-x-4">
    <div><ng-content select="app-avatar" /></div>
    <div>
        <ng-content select="[h4, p]" />
    </div>
</div>
