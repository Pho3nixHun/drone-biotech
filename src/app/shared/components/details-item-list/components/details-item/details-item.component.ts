import { Component } from '@angular/core';

@Component({
    selector: 'app-details-item',
    imports: [],
    templateUrl: './details-item.component.html',
})
export class DetailsItemComponent {}
