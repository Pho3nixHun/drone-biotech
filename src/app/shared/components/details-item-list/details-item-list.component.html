<div class="w-full flex flex-col space-y-4">
    <ng-content select="app-details-item" />
</div>
