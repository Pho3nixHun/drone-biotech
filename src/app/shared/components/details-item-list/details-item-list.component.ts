import { Component } from '@angular/core';

/**
 * DetailsItemListComponent
 *
 * Type: Presentational
 *
 * Scope:
 * - Responsible for rendering a list of details items.
 * - Manages the layout and styling of the details item list.
 *
 * Out-of-Scope:
 * - Should not handle the data fetching or business logic related to details item details.
 * - Should not be responsible for manipulating or transforming details item data.
 *
 * Purpose(optional):
 * - To create a simple layout that contains `DetailsItem`-s.
 */
@Component({
    selector: 'app-details-item-list',
    imports: [],
    templateUrl: './details-item-list.component.html',
})
export class DetailsItemListComponent {}
