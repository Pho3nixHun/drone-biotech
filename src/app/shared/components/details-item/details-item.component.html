<div class="flex flex-row space-x-4">
    <div><ng-content select="div" /></div>
    <div>
        <ng-content select="[h3, p]" />
    </div>
</div>
