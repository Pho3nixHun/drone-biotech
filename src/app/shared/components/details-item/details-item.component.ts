import { Component } from '@angular/core';

/**
 * DetailsItemComponent
 *
 * Type: Presentational
 *
 * Scope:
 * - Responsible for rendering an individual details item.
 *
 * Out-of-Scope:
 * - Should not handle data fetching.
 * - Contains no business logic related to the item.
 * - Does not manage or transform the item's data.
 *
 * Purpose (optional):
 * To visually present an individual details item.
 */
@Component({
    selector: 'app-details-item',
    imports: [],
    templateUrl: './details-item.component.html',
})
export class DetailsItemComponent {}
