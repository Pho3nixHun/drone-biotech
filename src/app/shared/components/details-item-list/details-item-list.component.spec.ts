import { ComponentFixture, TestBed } from '@angular/core/testing';
import { DetailsItemListComponent } from './details-item-list.component';
import { Component } from '@angular/core';
import { DetailsItemComponent } from '@components/details-item/details-item.component';

@Component({
    imports: [DetailsItemListComponent, DetailsItemComponent],
    template: `
        <app-details-item-list>
            <app-details-item></app-details-item>
            <div>Should not be projected</div>
        </app-details-item-list>
    `,
})
class TestHostComponent {}

describe('DetailsItemListComponent', () => {
    let fixture: ComponentFixture<TestHostComponent>;
    let compiled: HTMLElement;

    beforeEach(async () => {
        await TestBed.configureTestingModule({
            imports: [TestHostComponent],
        }).compileComponents();

        fixture = TestBed.createComponent(TestHostComponent);
        compiled = fixture.debugElement.nativeElement;
    });

    // Snapshot testing
    it('should project the content and render the template correctly', () => {
        // There is no need to arrange

        // Act
        fixture.detectChanges();

        // Assert
        expect(compiled.innerHTML).toMatchSnapshot();
    });
});
