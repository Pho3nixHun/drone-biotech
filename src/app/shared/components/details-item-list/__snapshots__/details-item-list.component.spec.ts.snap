// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DetailsItemListComponent should project the content and render the template correctly 1`] = `"<app-details-item-list><div class="w-full flex flex-col space-y-4"><app-details-item><div class="flex flex-row space-x-4"><div></div><div></div></div></app-details-item></div></app-details-item-list>"`;
